src/machine/aot/aot.x64.c

```diff
@@ -341,6 +342,76 @@ int aot_link(AotContext* context, size_t *szp)
   | mov rdx, CKB_VM_ASM_RET_INVALID_PERMISSION
   | pop rsi
   | ret
+  /*
+   * Fill the specified frame with zeros. Required arguments to this
+   * pseudo function include:
+   *
+   * rax: index of the frame, no overflow check.
+   */
+  |->zeroed_memory:
+  | push rdi
+  | push rsi
+  | mov rsi, rax
+  | shl rsi, CKB_VM_ASM_MEMORY_FRAME_SHIFTS
+  | lea rcx, machine->memory
+  | add rsi, rcx
+  | xor eax, eax
+  | mov ecx, CKB_VM_ASM_MEMORY_FRAMESIZE
+  | shr ecx, 2
+  | mov rdi, rsi
+  | cld
+  | rep; stosd
+  | pop rsi
+  | pop rdi
+  | ret
+  /*
+   * Zeroed frame by memory address and length if it's necessary.
+   *
+   * rax: the memory address to read/write
+   * rdx: length of memory to read/write
+   */
+  |->inited_memory:
+  | push rsi
+  | mov rsi, rdx
+  | mov rcx, rax
+  | shr rcx, CKB_VM_ASM_MEMORY_FRAME_SHIFTS
+  | cmp rcx, CKB_VM_ASM_MEMORY_FRAMES
+  | jae >2
+  | lea rdx, machine->frames
+  | movzx edx, byte [rdx+rcx]
+  | cmp edx, 1
+  | je >1
+  | lea rdx, machine->frames
+  | mov byte [rdx+rcx], 1
+  | push rax
+  | mov rax, rcx
+  | call ->zeroed_memory
```

Comment:
I feel like it's better we leverage memset than hand writing `zeroed_memory` function. Modern memset would leverage SIMD for more speedups.

Reply:
I do feel like it might be better we add a new [exit code](https://github.com/nervosnetwork/ckb-vm/blob/develop/definitions/src/asm.rs#L10-L16) here, when an uninitialized frame is detected, we exit from AOT and ASM mode with the error code, then at [Rust side](https://github.com/nervosnetwork/ckb-vm/blob/develop/src/machine/asm/mod.rs#L272), we can fill in the zeros using code which is just a memset. That to me, might be a better and (hopefully) faster way.

Reply:
I replaced it with `memset()`. They are called differently on different operating systems, and it took me a long time to understand this. It should work well for now.